docs/src/main/asciidoc/rest-client-reactive.adoc

```diff
@@ -858,11 +858,119 @@ quarkus.rest-client.logging.body-limit=50
 quarkus.log.category."org.jboss.resteasy.reactive.client.logging".level=DEBUG
 ----
 
+== Mocking the client for tests
+If you use a client injected with the `@RestClient` annotation, you can easily mock it for tests.
+You can do it with Mockito's `@InjectMock` or with `QuarkusMock`.
+
+This section shows how to replace your client with a mock. If you would like to get a more in-depth understanding of how mocking works in Quarkus, see the blog post on https://quarkus.io/blog/mocking/[Mocking CDI beans].
+
+NOTE: Mocking does not work when using `@NativeImageTest` or `@QuarkusIntegrationTest`.
+
+Let's assume you have the following client:
+[source,java]
+----
+package io.quarkus.it.rest.client.main;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+
+
+@Path("/")
+@RegisterRestClient
+public interface Client {
+    @GET
+    String get();
+}
+----
+
+
+=== Mocking with InjectMock
+The simplest approach to mock a client for tests is to use Mockito and `@InjectMock`.
+
+First, add the following dependency to your application:
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-junit5-mockito</artifactId>
+    <scope>test</scope>
+</dependency>
+----
+
+Then, in your test you can simply use `@InjectMock` to create and inject a mock:
+
+[source,java]
+----
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.mockito.InjectMock;
+
+@QuarkusTest
+public class InjectMockTest {
+
+    @InjectMock
+    ClientWithExceptionMapper mock;
```

Comment:
Where does the `ClientWithExceptionMapper` come from?

Reply:
oops, fixing